### PR TITLE
adding call to purchase order endpoint

### PR DESCRIPTION
**Pull request by edg205**

### CODE REVIEW

File: spec/support/vcr.rb [hunk 1]
```diff
@@ -1,10 +1,10 @@
 VCR.configure do |config|
-  config.cassette_library_dir = "spec/cassettes"
-  config.hook_into :faraday
-  config.configure_rspec_metadata!
-  config.allow_http_connections_when_no_cassette = false
-
-  config.default_cassette_options = {
-    record: :none
-  }
+  # config.cassette_library_dir = "spec/cassettes"
```
Comment (jeremyw) [inline spec/support/vcr.rb]: Did you intend to keep this?

File: spec/fosdick/purchase_order_spec.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  end
+
+  context "given an invalid purchase_order" do
+    it "POSTs the purchase order to Fosdick", vcr: { record: :none, cassette_name: "purchase_orders/create#invalid" } do
+      @attributes.delete(:qty)
+      result = Fosdick::PurchaseOrder.new(@attributes).create
+      expect(JSON.parse(result.body)).to match({"error"=>"qty must be provided"})
+    end
+  end
+
```
Comment (jeremyw) [inline spec/fosdick/purchase_order_spec.rb]: Let's clean up the whitespace here.

File: lib/fosdick/purchase_order.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    attribute :warehouse, String
+    attribute :expected_delivery, Date
+
+    def create
+      response = connection.post do |req|
+        req.url  "/fosdickapi/purchaseorder.json"
+        req.headers['Content-Type'] = 'application/json'
+        req.body = attributes.to_json
+      end
+      response
```
Comment (jeremyw) [inline lib/fosdick/purchase_order.rb]: Don't need the response variable.

File: spec/support/vcr.rb (hunk 1 above)
Comment (edg205) [inline spec/support/vcr.rb]: commented it out during testing, forgot to put it back in, doing that now

File: lib/fosdick/line_item.rb [hunk 4]
```diff
@@ -1,11 +1,11 @@
...
   class LineItem
-    include Virtus.model
+    include Virtus.model(:nullify_blank => true)
 
     attribute :inv, String
     attribute :qty, Integer
     attribute :price_per, Money
     attribute :num_of_payments, Integer, default: 1
-    attribute :line_custom, String
+    attribute :line_custom, String, default: ""
```
Comment (jeremyw) [inline lib/fosdick/line_item.rb]: Should the default be an empty string if we nullify blank strings?

File: spec/fosdick/order_spec.rb [hunk 5]
```diff
@@ -4,8 +4,8 @@
   before do
     @attributes = {
       email: "test@test.net",
-      external_id: "2316",
-      subtotal: 3.0,
+      external_id: "5584",
```
Comment (jeremyw) [inline spec/fosdick/order_spec.rb]: Why did this test need to be updated?

**Comment by jeremyw** [comment]
lgtm, let's squash and merge

File: lib/fosdick/purchase_order.rb [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    attribute :qty, Integer
+    attribute :warehouse, String
+    attribute :expected_delivery, Date
+    attribute :po_date, Date
+    attribute :vendor, String
+    attribute :shipper, String
+    attribute :upc, String
+    attribute :cogs, Integer
+    attribute :country_of_origin, String
+    attribute :weight, Integer
```
Comment (jeremyw) [inline lib/fosdick/purchase_order.rb]: These metrics may need to be Floats.

Comment (edg205) [inline lib/fosdick/purchase_order.rb]: and possibly cogs, will update

File: spec/fosdick/order_spec.rb (hunk 5 above)
Comment (jeremyw) [inline spec/fosdick/order_spec.rb]: Open question here

Comment (edg205) [inline spec/fosdick/order_spec.rb]: the test started to break when I pushed up my branch. something got messed up with the vcr cassette. I tried to regenerate the cassette and it show some issues so I fixed them. When we added the extra line items we didn't update the subtotal and total. there was also an issue with the connection.

**Pull request merged**